sonar-db/src/main/java/org/sonar/db/DatabaseChecker.java

```diff
@@ -58,29 +77,49 @@ public void stop() {
     // nothing to do
   }
 
-  private void checkOracleVersion() throws SQLException {
-    Connection connection = db.getDataSource().getConnection();
-    try {
-      // check version of db
-      // See http://jira.sonarsource.com/browse/SONAR-6434
-      int majorVersion = connection.getMetaData().getDatabaseMajorVersion();
-      if (majorVersion < ORACLE_MIN_MAJOR_VERSION) {
-        throw MessageException.of(String.format(
-          "Unsupported Oracle version: %s. Minimal required version is %d.", connection.getMetaData().getDatabaseProductVersion(), ORACLE_MIN_MAJOR_VERSION));
+  private void checkMinDatabaseVersion() throws SQLException {
+    Version minDbVersion = MINIMAL_SUPPORTED_DB_VERSIONS.get(db.getDialect().getId());
+    if (minDbVersion != null) {
+      try (Connection connection = db.getDataSource().getConnection()) {
+        int dbMajorVersion = connection.getMetaData().getDatabaseMajorVersion();
+        int dbMinorVersion = connection.getMetaData().getDatabaseMinorVersion();
+        Version dbVersion = new Version(dbMajorVersion, dbMinorVersion);
+        if (!dbVersion.isGreaterThanOrEqual(minDbVersion)) {
+          throw MessageException.of(String.format(
+            "Unsupported %s version: %s. Minimal supported version is %s.", db.getDialect().getId(), dbVersion, minDbVersion));
+        }
       }
+    }
+  }
 
-      // check version of driver
+  private void checkOracleDriverVersion() throws SQLException {
+    try (Connection connection = db.getDataSource().getConnection()) {
       String driverVersion = connection.getMetaData().getDriverVersion();
       String[] parts = StringUtils.split(driverVersion, ".");
       int intVersion = Integer.parseInt(parts[0]) * 100 + Integer.parseInt(parts[1]);
       if (intVersion < 1102) {
         throw MessageException.of(String.format(
-          "Unsupported Oracle JDBC driver version: %s. Minimal required version is 11.2.", driverVersion));
+          "Unsupported Oracle driver version: %s. Minimal supported version is 11.2.", driverVersion));
       }
-
-    } finally {
-      DbUtils.closeQuietly(connection);
     }
   }
 
+  private static class Version {
+    private final int major;
+    private final int minor;
+
+    public Version(int major, int minor) {
+      this.major = major;
+      this.minor = minor;
+    }
+
+    public boolean isGreaterThanOrEqual(Version other) {
+      return major >= other.major && (major != other.major || minor >= other.minor);
+    }
+
+    @Override
+    public String toString() {
+      return String.format("%d.%d", major, minor);
```

Comment:
as a general practice, `String.format` should be avoided in `toString()` methods, this method can be intensively called

It's not really the case, but here `major + '.' + minor` wouldn't be much longer to write nor harder to read

or did I miss a benefit of `String.format` in this case?